app-crypt/p11-kit/p11_kit-0.23.9.recipe

```diff
@@ -0,0 +1,93 @@
+SUMMARY="Provides a way to load and enumerate PKCS#11 modules"
+DESCRIPTION="p11-kit provides a way to load and enumerate PKCS#11 modules. \
+Provides a standard configuration setup for installing PKCS#11 modules \
+in such a way that they're discoverable."
+HOMEPAGE="https://p11-glue.freedesktop.org/p11-kit.html"
+COPYRIGHT="2017 Red Hat, Inc."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/p11-glue/p11-kit/releases/download/$portVersion/p11-kit-$portVersion.tar.gz"
+CHECKSUM_SHA256="e1c1649c335107a8d33cf3762eb7f57b2d0681f0c7d8353627293a58d6b4db63"
+SOURCE_DIR="p11-kit-$portVersion"
+PATCHES="p11_kit-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	p11_kit$secondaryArchSuffix = $portVersion
+	cmd:p11_kit$secondaryArchSuffix = $portVersion
+	cmd:trust$secondaryArchSuffix = $portVersion
+	lib:libp11_kit$secondaryArchSuffix = $portVersion
+	lib:p11_kit_proxy$secondaryArchSuffix = $portVersion
+	"
+
+# TODO: decide if ca_root_certificates is actually necessary
+REQUIRES="
+	haiku$secondaryArchSuffix
+	ca_root_certificates
+	lib:libffi$secondaryArchSuffix
+	lib:libtasn1$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	p11_kit${secondaryArchSuffix}_devel = $portVersion
+	devel:libp11_kit$secondaryArchSuffix = $portVersion
+	devel:p11_kit_proxy$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	p11_kit$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libffi$secondaryArchSuffix
+	devel:libtasn1$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:awk
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	"
+
+USER_SETTINGS_FILES="
+	settings/pkcs11 directory
+	"
+
+GLOBAL_WRITABLE_FILES="
+	settings/pkcs11/pkcs11.conf.example keep-old
+	"
+
+BUILD()
+{
+	certsInstallDir="$portPackageLinksDir/ca_root_certificates"
+	certsDir="$certsInstallDir/$relativeDataRootDir/ssl"
+	runConfigure ./configure \
+		LDFLAGS="-lnetwork" \
+		--with-trust-paths="${certsDir}/CARootCertificates.pem" \
+		--with-user-config="`finddir B_USER_SETTINGS_DIRECTORY`/pkcs11"
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	# clean up
+	rm -f $libDir/*.la $libDir/pkcs11/*.la
+
+	prepareInstalledDevelLibs \
+		libp11-kit \
```

Comment:
```diff
 INSTALL()
 {
 	make install
 
+	# Replace, in the config example, the real path of the settings dir by
+	# the user-friendly path. Then install it as the config file.
+	sed -i \
+		-e "s|$sysconfDir|`finddir B_SYSTEM_SETTINGS_DIRECTORY`|;" \
+		$settingsDir/pkcs11/pkcs11.conf.example
+	install -d $dataDir/pkcs11
+	install -t $dataDir/pkcs11 $settingsDir/pkcs11/pkcs11.conf.example
+	mv \
+		$settingsDir/pkcs11/pkcs11.conf.example \
+		$settingsDir/pkcs11/pkcs11.conf
+
 	# clean up
 	rm -f $libDir/*.la $libDir/pkcs11/*.la
 
+	# fix incorrect install directory on secondary arch builds
+	if [ -n "$secondaryArchSuffix" ]; then
+		mv $prefix/lib/p11-kit $libDir/p11-kit
+	fi
+
-	prepareInstalledDevelLibs \
-		libp11-kit \
-		p11-kit-proxy
+	prepareInstalledDevelLib libp11-kit
 	fixPkgconfig
 
 	packageEntries devel \
 		$developDir
 }
```
Notes: `p11-kit-proxy.so` is just a symlink to `libp11-kit.so.0.3.0`. We only need to provide that symlink in the base package. In the devel package it is (probably) not needed. So we don't need to pass it to `prepareInstalledDevelLib`.